x/merkledb/db.go

```diff
@@ -236,53 +231,47 @@ func newDatabase(
 	}
 
 	// mark that the db has not yet been cleanly closed
-	err = trieDB.metadataDB.Put(cleanShutdownKey, didNotHaveCleanShutdown)
+	err = trieDB.baseDB.Put(cleanShutdownKey, didNotHaveCleanShutdown)
 	return trieDB, err
 }
 
 // Deletes every intermediate node and rebuilds them by re-adding every key/value.
 // TODO: make this more efficient by only clearing out the stale portions of the trie.
 func (db *merkleDB) rebuild(ctx context.Context) error {
 	db.root = newNode(nil, RootPath)
-	if err := db.nodeDB.Delete(rootKey); err != nil {
-		return err
-	}
-	it := db.nodeDB.NewIterator()
-	defer it.Release()
 
-	viewSizeLimit := math.Max(
-		db.nodeCache.maxSize/rebuildViewSizeFractionOfCacheSize,
+	opsSizeLimit := math.Max(
+		db.valueNodeDB.nodeCache.maxSize/rebuildViewSizeFractionOfCacheSize,
 		minRebuildViewSizePerCommit,
 	)
-	currentOps := make([]database.BatchOp, 0, viewSizeLimit)
+	intermediateNodeIt := db.baseDB.NewIteratorWithPrefix(intermediateNodePrefix)
+	defer intermediateNodeIt.Release()
+	if err := database.ClearPrefix(db.baseDB, db.baseDB, intermediateNodePrefix); err != nil {
```

Comment:
will switch to new implementation when [1931](https://github.com/ava-labs/avalanchego/pull/1931) is merged